metaflow/plugins/aip/argo_utils.py

```diff
@@ -12,155 +12,290 @@
 logger = _get_aip_logger()
 
 
-def run_argo_workflow(
-    kubernetes_namespace: str,
-    template_name: str,
-    parameters: Optional[dict] = None,
-    wait_timeout: Union[int, float, datetime.timedelta] = 0,
-    **kwarg,  # Other parameters for wait function
-) -> Tuple[str, str]:
-    try:
-        # TODO(talebz): add tag of origin-run-id to correlate parent flow
-        workflow_manifest: Dict[str, Any] = ArgoClient(
-            namespace=kubernetes_namespace,
-        ).trigger_workflow_template(template_name, parameters)
-    except Exception as e:
-        raise AIPException(str(e))
-
-    argo_run_id = workflow_manifest["metadata"]["name"]
-    argo_run_uid = workflow_manifest["metadata"]["uid"]
-
-    if wait_timeout:  # int, float and datetime.timedelta all evaluates to False when 0
-        wait_for_argo_run_completion(
-            run_id=argo_run_id,
-            kubernetes_namespace=kubernetes_namespace,
-            wait_timeout=wait_timeout,
-            **kwarg,
-        )
+class ArgoHelper:
+    def __init__(self, kubernetes_namespace: str):
+        """
+        Args:
+            kubernetes_namespace: Namespace where Argo is running.
+                Required as the defaults provided in the ArgoClient is usually not what customers desire.
+                TODO: This namespace can be default to the current namespace if the script is ran within a cluster.
+        """
+        self._client = ArgoClient(namespace=kubernetes_namespace)
+
+    def template_submit(
```

Comment:
Could this be renamed to `template_trigger()`? 

My understanding is that this triggers a run of a workflow based off an existing workflow template. So it submits a request to trigger a run, but it does not submit a template, as the name implies (to me). What do you think?

Reply:
If you do change it here, also change it for the `template_submit_latest()` function.

Reply:
A second vote for `trigger`. Cool, I'll align with the metaflow command name so that it is more consistent to aip users.